presto-native-execution/presto_cpp/main/http/filters/JwtTokenFilter.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/http/filters/JwtTokenFilter.h"
+#include <folly/Range.h>
+#include <folly/io/IOBufQueue.h>
+#include <folly/ssl/OpenSSLHash.h>
+#include <proxygen/httpserver/ResponseBuilder.h>
+#include "presto_cpp/external/jwt-cpp/jwt.h"
+#include "presto_cpp/external/jwt-cpp/traits/nlohmann-json/traits.h"
+#include "presto_cpp/main/common/Configs.h"
+#include "presto_cpp/main/http/HttpConstants.h"
+
+namespace facebook::presto::http::filters {
+
+// If the internal communication is not enabled the filter
+// is not added to process incoming requests.
+void JWTTokenFilter::onRequest(
```

Comment:
Would you need to add any runtime counters in this logic ?

Reply:
Counter to count number of bad requests?
Not sure exactly what you mean. The Java code didn't seem to add counters so I didn't consider it.